src/Plugin/KeyProvider/PrivateFileKeyProvider.php

```diff
@@ -122,7 +122,7 @@ public function setKeyValue(KeyInterface $key, $key_value) {
     file_prepare_directory($file_path, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS);
     // Save the token data.
     if (!file_unmanaged_save_data($key_value, $file_uri, FILE_EXISTS_REPLACE)) {
-      throw new KeyValueNotSetException();
+      throw new KeyValueNotSetException('Error saving connection settings to file.');
```

Comment:
I'd also add the file path to the error message because that could be helpful in debugging, but this is probably just nitpicking.